api/migrations/20180501222559-add-isArchived-archivedBy.js

```diff
@@ -0,0 +1,31 @@
+exports.up = function(r, conn) {
+  /**
+    We're adding `isArchived` and defaulting it to `false` in `usersDirectMessageThreads`
+    All DMThreads are not archived
+  */
+  return Promise.all([
+    r
+      .table('usersDirectMessageThreads')
+      .update({
+        isArchived: false,
```

Comment:
Do you reckon we need to do this or can we let it be `undefined` for all existing threads and handle that as if it was `false`?

Reply:
Well, I chose `false` instead of `undefined` to keep things more _explicit_, but if you have strong opinions on `undefined` vs `false` I can change it.

How are handling this kind of flags across the whole application btw?

Reply:
That makes sense to me, that's exactly how we handle them elsewhere too. Let's do this!

Reply:
Re: comments on the PR - why don't we just leave `isArchived` off completely? The non-existence of the field already tells us that it's not archived, and avoids us having to write a migration :)

Reply:
Hmmm, that said (+ your [comment](https://github.com/withspectrum/spectrum/pull/2992#issuecomment-386027520))
`archivedAt` makes sense instead of `isArchived`, but `archivedBy` does not. We already have an `userId` in `usersDirectMessageThreads` so adding `archivedBy` will be duplicating information.